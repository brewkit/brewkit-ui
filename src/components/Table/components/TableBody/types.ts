import { ReactNode } from 'react';
import { ColumnPropsWithName, RowData } from '../TableRow/types';


export interface Props {


    /**
     * Any custom content to fill the TableBody that is different than default
     */
    children?: ReactNode,


    /**
     * The className for the TableBody
     */
    className?: string,


    /**
     * The configuration that defines the columns in the TableBody
     */
    columnConfig?: ColumnPropsWithName[],


    /**
     * The row data used to populate the TableBody
     */
    rows?: RowData[],


}
