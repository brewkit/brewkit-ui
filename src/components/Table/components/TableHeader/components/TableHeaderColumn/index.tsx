import React, { ReactElement } from 'react';
import clsx from 'clsx';
import TableCell from '../../../TableRow/components/TableCell';
import { Props } from './types';


/**
 * The `<TableHeaderColumn />` component represents a single column in the `<TableHeader />`.
 */
function TableHeaderColumn({
    children,
    className,
    label,
    name,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableHeaderColumn',
        className,
    );


    const content = children || label;


    return (
        <TableCell as="th" className={classes} {...otherProps}>
            {content}
        </TableCell>
    );


}


export default TableHeaderColumn;
