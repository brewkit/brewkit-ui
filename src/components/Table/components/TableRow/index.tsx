import React, { ReactElement, ReactNode } from 'react';
import clsx from 'clsx';
import TableCell from './components/TableCell';
import { Props } from './types';


/**
 * The `<TableRow />` component
 */
function TableRow({
    className,
    children,
    columnConfig,
    rowData,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableRow',
        className,
    );


    const content = children || (columnConfig?.map((col, index) => {
        if (rowData && col?.name && rowData[col.name]) {
            const cellData = rowData[col.name];
            if (cellData && typeof cellData === 'object' && 'label' in cellData) {
                return <TableCell key={index} {...cellData} />;
            }
            return (
                <TableCell key={index}>
                    {cellData}
                </TableCell>
            );
        }
    }));

    
    return (
        <tr className={classes} {...otherProps}>
            {content}
        </tr>
    );


}


TableRow.Cell = TableCell;


export default TableRow;
