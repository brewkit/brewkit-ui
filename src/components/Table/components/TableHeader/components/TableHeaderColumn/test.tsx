import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TableHeaderColumn from './index';
import Typography from '../../../../../Typography';


describe('Table Header Column', () => {


    it('Renders correctly with text children', () => {
        const header = renderer
            .create(<TableHeaderColumn>Label</TableHeaderColumn>)
            .toJSON();
        expect(header).toMatchSnapshot();
    });

    it('Renders correctly with component children', () => {
        const header = renderer
            .create(<TableHeaderColumn><Typography variant="h3">Label</Typography></TableHeaderColumn>)
            .toJSON();
        expect(header).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const header = shallow(<TableHeaderColumn className={className}>Label</TableHeaderColumn>);

        expect(header.hasClass(className)).toBe(true);
    });


});


describe('Table Header Column Props', () => {


    it('Renders correctly with text in `label` prop', () => {
        const header = renderer
            .create(<TableHeaderColumn label="Label1" />)
            .toJSON();
        expect(header).toMatchSnapshot();
    });


    it('Renders correctly with ReactNode in `label` prop', () => {
        const header = renderer
            .create(<TableHeaderColumn label={<Typography variant="h3">Label2</Typography>} />)
            .toJSON();

        expect(header).toMatchSnapshot();

        expect(shallow(<TableHeaderColumn label={<Typography variant="h3">Label2</Typography>} />)
            .find(Typography)
            .find('[variant="h3"]')
            .children()
            .contains('Label2')).toBe(true);
    });



});

