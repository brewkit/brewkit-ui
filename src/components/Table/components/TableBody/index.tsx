import React, { ReactElement, ReactNode } from 'react';
import clsx from 'clsx';
import { RowData } from '../TableRow/types';
import TableRow from '../TableRow';
import { Props } from './types';


/**
 * The `<TableBody />` component represents the standard body of a `<Table />` component.
 */
function TableBody({
    className,
    children,
    columnConfig,
    rows,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableBody',
        className,
    );


    const content = children || (rows?.map((rowData: RowData, index: number) => (
        <TableRow columnConfig={columnConfig} key={index} rowData={rowData} />
    )));


    return (
        <tbody className={classes} {...otherProps}>
            {content}
        </tbody>
    );


}


export default TableBody;
