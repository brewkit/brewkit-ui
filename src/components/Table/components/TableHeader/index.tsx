import React, { ReactElement } from 'react';
import clsx from 'clsx';
import TableHeaderColumn from './components/TableHeaderColumn';
import { Props as ColProps } from './components/TableHeaderColumn/types';
import TableRow from '../TableRow';
import { Props } from './types';


/**
 * The `<TableHeader />` component represents the standard header of a `<Table />` component.
 */
function TableHeader({
    children,
    className,
    columnConfig,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableHeader',
        className,
    );


    const content = children || (columnConfig?.map((colData: ColProps, index: number) => (
        <TableHeaderColumn key={index} {...colData} />
    )));


    return (
        <thead className={classes} {...otherProps}>
            <TableRow>
                {content}
            </TableRow>
        </thead>
    );


}


TableHeader.Column = TableHeaderColumn;


export default TableHeader;
