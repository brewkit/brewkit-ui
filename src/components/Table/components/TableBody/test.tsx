import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TableBody from './index';
import Typography from '../../../Typography';


describe('Table Body', () => {


    it('Renders correctly', () => {
        const content = renderer
            .create((
                <TableBody>
                    <tr>
                        <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>
                    </tr>
                    <tr>
                        <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                    </tr>
                </TableBody>
            ))
            .toJSON();

        expect(content).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow((
            <table>
                <TableBody className={className}>
                    <tr>
                        <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>

                    </tr>
                    <tr>
                        <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                    </tr>
                </TableBody>
            </table>
        ));

        expect(content.children().first()
            .shallow()
            .hasClass(className))
            .toBe(true);
        expect(content.children().first()
            .shallow()
            .hasClass('brew-TableBody'))
            .toBe(true);
    });


});


describe('Table Body Props', () => {


    it('Renders `columnConfig` and `rowData` props correctly', () => {
        const content = renderer
            .create((
                <TableBody
                    columnConfig={[
                        { name: 'ColA' },
                        { name: 'ColB' },
                        { name: 'ColC' },
                    ]}
                    rows={[
                        {
                            ColA: 'Row1-ColA',
                            ColB: 'Row1-ColB',
                            ColC: 'Row1-ColC',
                        },
                        {
                            ColA: 'Row2-ColA',
                            ColB: 'Row2-ColB',
                            ColC: 'Row2-ColC',
                        },
                        {
                            ColA: 'Row3-ColA',
                            ColB: 'Row3-ColB',
                            ColC: 'Row3-ColC',
                        }
                    ]}
                />
            ))
            .toJSON();

        expect(content).toMatchSnapshot()

        expect(mount((
            <table>
                <TableBody
                    columnConfig={[
                        { name: 'ColD' },
                        { name: 'ColE' },
                        { name: 'ColF' },
                    ]}
                    rows={[
                        {
                            ColD: <Typography variant="body1">Row1-ColD</Typography>,
                            ColE: <Typography variant="body1">Row1-ColE</Typography>,
                            ColF: <Typography variant="body1">Row1-ColF</Typography>,                    
                        },
                        {
                            ColD: <Typography variant="body1">Row2-ColD</Typography>,
                            ColE: <Typography variant="body1">Row2-ColE</Typography>,
                            ColF: <Typography variant="body1">Row2-ColF</Typography>,                    
                        },
                        {
                            ColD: <Typography variant="body1">Row3-ColD</Typography>,
                            ColE: <Typography variant="body1">Row3-ColE</Typography>,
                            ColF: <Typography variant="body1">Row3-ColF</Typography>,                    
                        }
                    ]}
                />
            </table>
        ))
            .find(Typography)
            .find('[variant="body1"]')
            .children()
            .contains('Row2-ColE')).toBe(true);
    });


});
