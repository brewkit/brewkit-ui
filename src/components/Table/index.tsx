import React, { ElementType, ReactElement } from 'react';
import clsx from 'clsx';
import TableBody from './components/TableBody';
import TableHeader from './components/TableHeader';
import TableRow from './components/TableRow';
import { Props } from './types';


/**
 * The `<Table />` component is used to build a responsive table.
 */
function Table({
    as,
    className,
    children,
    columnConfig,
    rows,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-Table',
        className,
    );


    const TableElement: ElementType<any> = as || 'table';


    const content = children || (
        <React.Fragment>
            <TableHeader columnConfig={columnConfig} />
            <TableBody columnConfig={columnConfig} rows={rows} />
        </React.Fragment>
    );


    return (
        <TableElement className={classes} {...otherProps}>
            {content}
        </TableElement>
    );


}


Table.Body = TableBody;
Table.Header = TableHeader;
Table.Row = TableRow;


export default Table;
