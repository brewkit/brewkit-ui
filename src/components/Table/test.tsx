import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Table from './index';
import TableBody from './components/TableBody';
import TableHeader from './components/TableHeader';
import TableHeaderColumn from './components/TableHeader/components/TableHeaderColumn';
import TableRow from './components/TableRow';
import Typography from '../Typography';


describe('Table', () => {


    it('Renders correctly', () => {
        const content = renderer
            .create((
                <Table>
                    <thead>
                        <th>Column1</th>
                        <th>Column2</th>
                        <th>Column3</th>
                    </thead>
                    <tbody>
                        <tr>
                            <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>
                        </tr>
                        <tr>
                            <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                        </tr>
                    </tbody>
                </Table>
            ))
            .toJSON();
        expect(content).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow((
            <Table className={className}>
                <thead>
                    <th>Column1</th>
                    <th>Column2</th>
                    <th>Column3</th>
                </thead>
                <tbody>
                    <tr>
                        <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>
                    </tr>
                    <tr>
                        <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                    </tr>
                </tbody>
            </Table>
        ));

        expect(content.hasClass(className)).toBe(true);
        expect(content.hasClass('brew-Table')).toBe(true);
    });


});

describe('Table Props - `as`', () => {


    it('Renders `as` prop', () => {
        const content = renderer
            .create((
                <Table
                    as="div"
                    children="Hello"
                />
            ))
            .toJSON();
        expect(content).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow((
            <Table
                as="p"
                children={<span>Hello</span>}
                className={className}
            />
        ));

        expect(content.hasClass(className)).toBe(true);
        expect(content.hasClass('brew-Table')).toBe(true);
    });


});


describe('Table Props - `columnConfig` only', () => {


    it('Renders `columnConfig` prop correctly (`label` and `name`)', () => {
        const content = renderer
            .create((
                <Table
                    columnConfig={[
                        {
                            label: 'ColumnA',
                            name: 'ColA',
                        },
                        {
                            label: 'ColumnB',
                            name: 'ColB',
                        },
                        {
                            label: 'ColumnC',
                            name: 'ColC',
                        },
                    ]}
                />
            ))
            .toJSON();
        expect(content).toMatchSnapshot();

        const mounted = mount(
            <Table
                columnConfig={[
                    {
                        label: <Typography variant="body1">ColumnA</Typography>,
                        name: 'ColA',
                    },
                    {
                        label: <Typography variant="body1">ColumnB</Typography>,
                        name: 'ColB',
                    },
                    {
                        label: <Typography variant="body1">ColumnC</Typography>,
                        name: 'ColC',
                    },
                ]}
            />
        );

        expect(mounted
            .find(Typography)
            .find('[variant="body1"]')
            .children()
            .contains('ColumnA')).toBe(true);
    });


});


describe('Table Props - `columnConfig` and `rows`', () => {


    it('Renders `rows` prop (without objects) correctly', () => {
        const content = renderer
            .create((
                <Table
                    columnConfig={[
                        { name: 'ColA' },
                        { name: 'ColB' },
                        { name: 'ColC' },
                    ]}
                    rows={[
                        {
                            ColA: 'Row1-ColA',
                            ColB: 'Row1-ColB',
                            ColC: 'Row1-ColC',
                        },
                        {
                            ColA: 'Row2-ColA',
                            ColB: 'Row2-ColB',
                            ColC: 'Row2-ColC',
                        },
                        {
                            ColA: 'Row3-ColA',
                            ColB: 'Row3-ColB',
                            ColC: 'Row3-ColC',
                        },
                    ]}
                />
            ))
            .toJSON();
        expect(content).toMatchSnapshot();

        expect(mount((
            <Table
                columnConfig={[
                    {
                        label: <Typography variant="h3">ColumnX</Typography>,
                        name: 'ColX',
                    },
                    {
                        label: <Typography variant="h3">ColumnY</Typography>,
                        name: 'ColY',
                    },
                    {
                        label: <Typography variant="h3">ColumnZ</Typography>,
                        name: 'ColZ',
                    },
                ]}
                rows={[
                    {
                        ColX: <Typography key="row1-colx" variant="body2">Row1-ColX</Typography>,
                        ColY: <Typography key="row1-coly" variant="body2">Row1-ColY</Typography>,
                        ColZ: <Typography key="row1-colz" variant="body2">Row1-ColZ</Typography>,
                    },
                    {
                        ColX: <Typography key="row2-colx" variant="body2">Row2-ColX</Typography>,
                        ColY: <Typography key="row2-coly" variant="body2">Row2-ColY</Typography>,
                        ColZ: <Typography key="row2-colz" variant="body2">Row2-ColZ</Typography>,
                    },
                    {
                        ColX: <Typography key="row3-colx" variant="body2">Row3-ColX</Typography>,
                        ColY: <Typography key="row3-coly" variant="body2">Row3-ColY</Typography>,
                        ColZ: <Typography key="row3-colz" variant="body2">Row3-ColZ</Typography>,
                    },
                ]}
            />
        ))
            .find(Typography)
            .find('[variant="body2"]')
            .children()
            .contains('Row2-ColY')).toBe(true);
    });


    it('Renders `rows` prop (with objects) correctly', () => {
        const content = renderer
            .create((
                <Table
                    columnConfig={[
                        { name: 'ColA' },
                        { name: 'ColB' },
                        { name: 'ColC' },
                    ]}
                    rows={[
                        {
                            ColA: 'Row1-ColA',
                            ColB: {
                                colspan: 2,
                                label: 'Row1-ColB&ColC',
                            },
                        },
                        {
                            ColA: 'Row2-ColA',
                            ColB: 'Row2-ColB',
                            ColC: 'Row2-ColC',
                        },
                    ]}
                />
            ))
            .toJSON();
        expect(content).toMatchSnapshot();

        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const mounted = mount(
            <Table
                columnConfig={[
                    {
                        label: <Typography variant="h3">ColumnY</Typography>,
                        name: 'ColY',
                    },
                    {
                        label: <Typography variant="h3">ColumnZ</Typography>,
                        name: 'ColZ',
                    },
                ]}
                rows={[
                    {
                        ColX: <Typography key="row2-colx" variant="body2">Row1-ColX</Typography>,
                        ColY: {
                            className: className,
                            label: <Typography key="row2-coly" variant="body2">Row1-ColY</Typography>,
                        },
                        ColZ: <Typography key="row2-colz" variant="body2">Row1-ColZ</Typography>,
                    },
                    {
                        ColX: <Typography key="row3-colx" variant="body2">Row2-ColX</Typography>,
                        ColY: <Typography key="row3-coly" variant="body2">Row2-ColY</Typography>,
                        ColZ: <Typography key="row3-colz" variant="body2">Row2-ColZ</Typography>,
                    },
                ]}
            />
        );

        expect(mounted
            .find(Typography)
            .find('[variant="body2"]')
            .children()
            .contains('Row2-ColZ')).toBe(true);

        expect(mounted
            .find(`.${className}`)
            .children()
            .contains('Row1-ColY')).toBe(true);
    });


});


describe('Full Table Suite', () => {


    it('Renders correctly', () => {
        const content = renderer
            .create((
                <Table>
                    <TableHeader>
                        <TableHeaderColumn label='Column1' />
                        <TableHeaderColumn label='Column2' />
                        <TableHeaderColumn label='Column3' />
                    </TableHeader>
                    <TableBody>
                        <TableRow>
                            <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>
                        </TableRow>
                        <TableRow>
                            <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                        </TableRow>
                    </TableBody>
                </Table>
            ))
            .toJSON();

        expect(content).toMatchSnapshot();
    });


});