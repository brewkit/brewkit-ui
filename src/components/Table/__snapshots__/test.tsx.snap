// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full Table Suite Renders correctly 1`] = `
<table
  className="brew-Table"
>
  <thead
    className="brew-TableHeader"
  >
    <tr
      className="brew-TableRow"
    >
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        Column1
      </th>
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        Column2
      </th>
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        Column3
      </th>
    </tr>
  </thead>
  <tbody
    className="brew-TableBody"
  >
    <tr
      className="brew-TableRow"
    >
      <td>
        Row1-Col1
      </td>
      <td>
        Row1-Col2
      </td>
      <td>
        Row1-Col3
      </td>
    </tr>
    <tr
      className="brew-TableRow"
    >
      <td>
        Row2-Col1
      </td>
      <td>
        Row2-Col2
      </td>
      <td>
        Row2-Col3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Props - \`as\` Renders \`as\` prop 1`] = `
<div
  className="brew-Table"
>
  Hello
</div>
`;

exports[`Table Props - \`columnConfig\` and \`rows\` Renders \`rows\` prop (with objects) correctly 1`] = `
<table
  className="brew-Table"
>
  <thead
    className="brew-TableHeader"
  >
    <tr
      className="brew-TableRow"
    >
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        
      </th>
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        
      </th>
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        
      </th>
    </tr>
  </thead>
  <tbody
    className="brew-TableBody"
  >
    <tr
      className="brew-TableRow"
    >
      <td
        className="brew-TableCell"
      >
        Row1-ColA
      </td>
      <td
        className="brew-TableCell"
        colspan={2}
      >
        Row1-ColB&ColC
      </td>
    </tr>
    <tr
      className="brew-TableRow"
    >
      <td
        className="brew-TableCell"
      >
        Row2-ColA
      </td>
      <td
        className="brew-TableCell"
      >
        Row2-ColB
      </td>
      <td
        className="brew-TableCell"
      >
        Row2-ColC
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Props - \`columnConfig\` and \`rows\` Renders \`rows\` prop (without objects) correctly 1`] = `
<table
  className="brew-Table"
>
  <thead
    className="brew-TableHeader"
  >
    <tr
      className="brew-TableRow"
    >
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        
      </th>
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        
      </th>
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        
      </th>
    </tr>
  </thead>
  <tbody
    className="brew-TableBody"
  >
    <tr
      className="brew-TableRow"
    >
      <td
        className="brew-TableCell"
      >
        Row1-ColA
      </td>
      <td
        className="brew-TableCell"
      >
        Row1-ColB
      </td>
      <td
        className="brew-TableCell"
      >
        Row1-ColC
      </td>
    </tr>
    <tr
      className="brew-TableRow"
    >
      <td
        className="brew-TableCell"
      >
        Row2-ColA
      </td>
      <td
        className="brew-TableCell"
      >
        Row2-ColB
      </td>
      <td
        className="brew-TableCell"
      >
        Row2-ColC
      </td>
    </tr>
    <tr
      className="brew-TableRow"
    >
      <td
        className="brew-TableCell"
      >
        Row3-ColA
      </td>
      <td
        className="brew-TableCell"
      >
        Row3-ColB
      </td>
      <td
        className="brew-TableCell"
      >
        Row3-ColC
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Props - \`columnConfig\` only Renders \`columnConfig\` prop correctly (\`label\` and \`name\`) 1`] = `
<table
  className="brew-Table"
>
  <thead
    className="brew-TableHeader"
  >
    <tr
      className="brew-TableRow"
    >
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        ColumnA
      </th>
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        ColumnB
      </th>
      <th
        className="brew-TableCell brew-TableHeaderColumn"
      >
        ColumnC
      </th>
    </tr>
  </thead>
  <tbody
    className="brew-TableBody"
  />
</table>
`;

exports[`Table Renders correctly 1`] = `
<table
  className="brew-Table"
>
  <thead>
    <th>
      Column1
    </th>
    <th>
      Column2
    </th>
    <th>
      Column3
    </th>
  </thead>
  <tbody>
    <tr>
      <td>
        Row1-Col1
      </td>
      <td>
        Row1-Col2
      </td>
      <td>
        Row1-Col3
      </td>
    </tr>
    <tr>
      <td>
        Row2-Col1
      </td>
      <td>
        Row2-Col2
      </td>
      <td>
        Row2-Col3
      </td>
    </tr>
  </tbody>
</table>
`;
